src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -351,7 +351,7 @@ protected function instantiateObject(array &$data, string $class, array &$contex
             $params = [];
             foreach ($constructorParameters as $constructorParameter) {
                 $paramName = $constructorParameter->name;
-                $key = $this->nameConverter ? $this->nameConverter->normalize($paramName, $class, $format, $context) : $paramName;
+                $key = $this->nameConverter ? $this->nameConverter->denormalize($paramName, $class, $format, $context) : $paramName;
```

Comment:
Not sure about your issue, but this bugs me out:
I think the code was correct ; for each of the constructor properties, it attempts to get the normalized name, to use as key in the `$data` array.

Calling `denormalize` would have been done the other way around: starting from `$data` keys, call `denormalize` on each to get the property name.

Reply:
You were right, the test proves that this code is correct (if the input is not already converted).